api/src/main/java/com/datastrato/gravitino/exceptions/GravitinoRuntimeException.java

```diff
@@ -17,12 +20,25 @@ public GravitinoRuntimeException(String message) {
   }
 
   /**
-   * Constructs a new exception with the specified detail message and cause.
+   * Constructs a new exception with the specified detail message.
    *
    * @param message the detail message.
+   * @param args the arguments to the message.
+   */
+  @FormatMethod
+  public GravitinoRuntimeException(@FormatString String message, Object... args) {
+    super(String.format(message, args));
+  }
+
+  /**
+   * Constructs a new exception with the specified detail message and cause.
+   *
    * @param cause the cause.
+   * @param message the detail message.
+   * @param args the arguments to the message.
    */
-  public GravitinoRuntimeException(String message, Throwable cause) {
-    super(message, cause);
+  @FormatMethod
+  public GravitinoRuntimeException(Throwable cause, @FormatString String message, Object... args) {
```

Comment:
The `Throwable` must be the first argument. Or it will throw an exception. I think it is conflict with the variable parameters in the overloaded function
```
@FormatMethod
  public GravitinoRuntimeException(@FormatString String message, Object... args) {
    super(String.format(message, args));
  }
```